src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -309,24 +309,37 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 
 	private getCurrentSelectionText(): string {
 		if (this.cellModel.currentMode === CellEditModes.WYSIWYG) {
-			return document.getSelection()?.toString() || '';
+			return document.getSelection()?.anchorNode.nodeValue || '';
 		} else {
 			const editorControl = this.getCellEditorControl();
 			const selection = editorControl?.getSelection();
 			if (selection && !selection.isEmpty()) {
 				const textModel = editorControl?.getModel() as TextModel;
 				const value = textModel?.getValueInRange(selection);
-				return value || '';
+				let linkLabel = value.substring(value.indexOf('[') + 1, value.lastIndexOf(']'));
```

Comment:
I will look into refining this to regex function. I think the incremental step of focusing on one selected link similar to what WYSIWYG currently does is important. I was focused on ensuring that given a link in markdown format [](), then the display text would be anything that is contained in the [] and the address would be anything within the (). 

1. I will work on fixing the assumption to ensure that if not link then we appropriately place the title.
2. I'm a bit confused about what you mean about this. It will ensure that anything inside the brackets will be included, so even if a title had brackets it will still be included as well so long that there is a closing bracket.
![image](https://user-images.githubusercontent.com/23587151/115076094-f06c6380-9eb0-11eb-9b9a-a1bba95ec093.png)
3. Understanding what the overall design and functionality of this should be discussed first and if it is a scenario (live site users) typically do, and what they expect.
4.  Yes, this would break. Will look into this a bit more about using regex to verify it. Will focus on the scenario of selecting a valid link in markdown.